### DISCUSSION TITLE
Parsing through a CSV for specific delay values

### DISCUSSION THREAD

**Discussion by mcarleto** [bonsai-rx/bonsai#882] [Q&A] [answered]
Hi, 
I'm quite new to bonsai and am trying to set up a behavioral trial. I'm essentially stuck trying to read a CSV file that contains my inter-stimulus intervals. It seems to be reading the last number and not actually parsing through the CSV. I've looked at so many examples and can't find where I'm going wrong. I need set delays before my circle stimulus comes on for a set period of time.

I've attached a photo my bonsai layout.

Thanks!
Maya
<img width="689" alt="lightstim" src="https://user-images.githubusercontent.com/104100583/164311708-1a847a54-4c82-4f37-9c79-2d16eb360172.png">

<img width="461" alt="mainworkflow" src="https://user-images.githubusercontent.com/104100583/164311725-b0efbe45-fe58-454c-87ff-04644faa493b.png">

**Comment by bruno-f-cruz** [comment]
Hi and welcome to Bonsai :)

Can you add the workflow you are working on? thanks!

**Reply by mcarleto** [reply]
Hi Bruno, 

I've attached the zip file with the workflow and the csv I'm using. It will run with an initial delay before the circle comes on but instead of sequencing through the CSV for the rest of the delay values it continues to grab the same value. 

Thanks!

[420timingintervals.csv](https://github.com/bonsai-rx/bonsai/files/8531574/420timingintervals.csv)
[crashtest.bonsai.zip](https://github.com/bonsai-rx/bonsai/files/8531578/crashtest.bonsai.zip)

**Comment by bruno-f-cruz** [comment, marked as answer]
The easiest way to do this is probably to use 'CreateObservables' for each trial. Can you check if this is the behavior you are looking for?
[crashtest.zip](https://github.com/bonsai-rx/bonsai/files/8532140/crashtest.zip)

**Reply by mcarleto** [reply]
Wow that is so much simpler and cleaner and it does the job! Thanks so much for your help I really appreciate you guys taking the time to answer everyone's questions.

Best, 

Maya
